.github/workflows/release.yml

```
@@ -79,8 +79,12 @@ jobs:
 
   build-linux:
     name: Build Linux
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     steps:
+      - name: Install dev tools
+        run: | 
+          sudo apk update
+          sudo apk add build-base linux-headers zip libressl-dev
```

Should this be `apt`?

```suggestion
          sudo apt update
          sudo apt -y install build-base linux-headers zip libressl-dev
```

---

Also, why `musl-tools` and `libssl-dev` when building, but `build-base`, `linux-headers`, and `libressl-dev` here? Also #2, do you know why we need the kernel headers at this point?

---

In any case, I think it would help to add comments explaining why each dependency is being pulled in.

---

nice catch! I was going back and forth between alpine and ubuntu and forgot to update one of the flows.

---

as for the comments, it was really my best guest judging from various answers in the internet. Seems like rust compilation process is using musl gcc which is not available by default in ubuntu. My understanding is that the new libraries I included require custom compilation steps which trigger this compiler and linking with libssl.